Sources/IRGen/Function/IULIAStatement.swift

```diff
@@ -63,6 +63,123 @@ struct IULIAIfStatement {
   }
 }
 
+/// Generates code for a for statement.
+struct IULIAForStatement {
+  var forStatement: ForStatement
+
+  func rendered(functionContext: FunctionContext) -> String {
+    var functionContext = functionContext
+    functionContext.scopeContext = forStatement.forBodyScopeContext!
+    
+    let forSetup: String
+    
+    switch forStatement.iterable {
+    case .identifier(let iterableIdentifier):
+      // Iterating over an array
+      guard let offset = functionContext.environment.propertyOffset(for: iterableIdentifier.name, enclosingType: functionContext.enclosingTypeName) else {
+        fatalError("Invalid offset")
+      }
+      let type: Type.RawType
+      let storageArrayOffset: String
+      
+      let loadArrLen: String
+      type = functionContext.environment.type(of: iterableIdentifier.name, enclosingType: functionContext.enclosingTypeName)
+      
+      switch type {
+      case .arrayType(_):
+        storageArrayOffset = IULIARuntimeFunction.storageArrayOffset(arrayOffset: offset, index: "i")
+        loadArrLen = "sload(\(offset))"
+      case .fixedSizeArrayType(_):
+        let typeSize = functionContext.environment.size(of: type)
+        loadArrLen = String(typeSize)
+        storageArrayOffset = IULIARuntimeFunction.storageFixedSizeArrayOffset(arrayOffset: offset, index: "i", arraySize: typeSize)
+      default:
+        fatalError()
+      }
+      let subscriptLoad = IULIARuntimeFunction.load(address: storageArrayOffset, inMemory: false)
+      let variableUse = IULIAAssignment(lhs: .identifier(forStatement.variable.identifier), rhs: .rawAssembly(subscriptLoad, resultType: nil)).rendered(functionContext: functionContext, asTypeProperty: false)
+      
+      forSetup = """
+      {
+      let i := 0
+      let _arrLen := \(loadArrLen)
```

Comment:
This should really be `arrLen` rather than `_arrLen`, to avoid conflicts with user code.